torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,162 @@
+    qualified_name = torch.jit._qualified_name(module_interface)
+    cu = torch.jit._python_cu
+    module_interface_c = cu.get_interface(qualified_name)
+    assert (
+        "forward" in module_interface_c.getMethodNames()
+    ), "Expect forward in interface methods, while it has {}".format(
+        module_interface_c.getMethodNames()
+    )
+    method_schema = module_interface_c.getMethod("forward")
+
+    arg_str_list = []
+    arg_type_str_list = []
+    for argument in method_schema.arguments:
+        arg_str_list.append(argument.name)
+
+        if argument.has_default_value():
+            default_value_str = " = {}".format(argument.default)
+        else:
+            default_value_str = ""
+        arg_type_str = "{name}: {type}{default_value}".format(
+            name=argument.name, type=argument.type, default_value=default_value_str
+        )
+        arg_type_str_list.append(arg_type_str)
+
+    arg_str_list = arg_str_list[1:]  # Remove "self".
+    args_str = ", ".join(arg_str_list)
+
+    arg_type_str_list = arg_type_str_list[1:]  # Remove "self".
+    arg_types_str = ", ".join(arg_type_str_list)
+
+    assert len(method_schema.returns) == 1
+    argument = method_schema.returns[0]
+    return_type_str = str(argument.type)
+
+    return args_str, arg_types_str, return_type_str
+
+
+def cleanup_generated_modules():
+    generated_module_paths = pathlib.Path(INSTANTIATED_TEMPLATE_DIR_PATH).glob(
+        f"{_FILE_PREFIX}*.py"
+    )
+    for file_path in generated_module_paths:
+        try:
+            print(f"Removing {file_path}")
+            assert file_path.is_file(), f"Epect {file_path} to be a file"
+            file_path.unlink()
+        except Exception as exc:
+            logger.warning(f"Failed to remove {file_path}:\n{traceback.format_exc()}")
+
+
+def _write(out_path, text):
+    try:
+        with open(out_path, "r") as f:
+            old_text = f.read()
+    except IOError:
+        old_text = None
+    if old_text != text:
+        with open(out_path, "w") as f:
+            print("Writing {}".format(out_path))
+            f.write(text)
+    else:
+        print("Skipped writing {}".format(out_path))
+
+
+def _do_instantiate_remote_module_template(generated_module_name, str_dict):
+    generated_code_text = REMOTE_MODULE_TEMPLATE.format(**str_dict)
+    out_path = os.path.join(
+        INSTANTIATED_TEMPLATE_DIR_PATH, f"{generated_module_name}.py"
+    )
+    _write(out_path, generated_code_text)
+
+    # From importlib doc,
+    # > If you are dynamically importing a module that was created since
+    # the interpreter began execution (e.g., created a Python source file),
+    # you may need to call invalidate_caches() in order for the new module
+    # to be noticed by the import system.
+    importlib.invalidate_caches()
+    generated_module = importlib.import_module(
+        f"torch.distributed.nn.jit.templates.instantiated.{generated_module_name}"
```

Comment:
this template seems not only getting used by jit, also getting used by nn, so maybe don't put it in the jit foler?

Reply:
I put it separate because this TorchScript support is an extension to `nn.RemoteModule`.
If the created RemoteModule does not to be scripted, code in this file is not called.

Opening a separate folder because the template initiation tend to become more complex over time, if we want to support generating methods beyond "forward".